### ISSUE TITLE
Error when loading weights of pre-trained checkpoint

### ISSUE THREAD

**Issue by adithyamurali** [torch-points3d/torch-points3d#297]
Hi Team,
Thanks for the great repo! I was able to use the API to train PointNet2 on the ShapeNet dataset using the following command:
`python train.py task=segmentation model_type=pointnet2 model_name=pointnet2_charlesssg dataset=shapenet-fixed`

The script by default only happened to train the 'Cap' category, so I modified it to train all categories by setting the `use_category` variable to False and deleting the `category: "Cap"` variable in this [config file](https://github.com/nicolas-chaulet/torch-points3d/blob/master/conf/data/segmentation/shapenet-fixed.yaml#L7).

Now, to run this model for inference, I followed the steps in the README and made the following variables changes in this [config file](https://github.com/nicolas-chaulet/torch-points3d/blob/master/forward_scripts/conf/config.yaml) in the forward_scripts folder:
`checkpoint_dir: Location of model weights from previous training step`
`output_path: Location to output predictions`
`input_path: Location of shapenet test data`
`defaults-dataset: "shapenet-fixed"`

After this, I ran:
`python forward_scripts/forward.py`

This was the error I got:
![Screenshot from 2020-06-10 18-05-36](https://user-images.githubusercontent.com/4186687/84323598-14e70900-ab45-11ea-9a58-85b509dab510.png)

I'm not sure if i'm doing something fundamentally wrong or I'm missing some config parameters. Thanks!

**Comment by nicolas-chaulet** [comment]
Hi!
If you have trained on cap only you cannot do an inference on the whole dataset (labels have changed and therefore the last mlp is different). The loading of the checkpoint fails because the model you have built to run the inference on the whole dataset does not have the same shape as your weights. Does that make sense? In order to do what you wnt, you will need to train again but without Cap.

**Comment by nicolas-chaulet** [comment]
Sorry, it is something else actually. So you have trained on the whole dataset but the foward for some reason builds a model that is too small in its output layer. Let me take a look into it.

**Comment by tchaton** [comment]
use_category should be True. It indicates to the model we are doing hierarchical segmentation.
Try re-training with use_category: True and remove category from config file.

**Comment by adithyamurali** [comment]
Thanks @nicolas-chaulet  and @tchaton! I just tried training again by setting `use_category : True` and removing `category` from the config file. I still get the same error but the weight dimensions seem to have been changed slightly from before:

![Screenshot from 2020-06-11 09-13-36](https://user-images.githubusercontent.com/4186687/84389821-a21d7280-abc4-11ea-807b-b9e0a132f7e3.png)

**Comment by nicolas-chaulet** [comment]
Yes, I think it is specific to the foward.py script, we haven't maintained it lately. Have you tried to run the eval.py script?

**Comment by thzou** [comment]
![Screenshot from 2021-04-08 17-38-35](https://user-images.githubusercontent.com/18090070/114055705-4af23800-9891-11eb-86b0-21ae1ea39e2f.png)
I am experiencing nearly the same issues on forward scripts. Please find attached my screenshot. For clarification, the model is trained on shapenet-fixed dataset with the use_category=True and without specifying the category parameter. However, I am receiving this error when trying to use forward script for the 'Cap' category. Any solution?

**Comment by CCInc** [comment]
Closing due to being old, please open a new issue if needed
